Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLWriter.java

```diff
@@ -0,0 +1,352 @@
+                    }
+                } else {
+                    bagElements.put(relation, objects);
+                }
+            }
+        }
+
+        if ((first != Node.FIRST_INDEX) && !orderedElements.isEmpty()) {
+            throw new IndexOutOfBoundsException(
+                    "First element at illegal index " + first + ", must be " + Node.FIRST_INDEX);
+        }
+        if (orderedElements.size() != last) {
+            throw new IndexOutOfBoundsException(orderedElements.size() + " elements found, but " + last + " expected.");
+        }
+
+        for (Entry<Long, ObjectType> entry : orderedElements.entrySet()) {
+            ObjectType object = entry.getValue();
+            if (object instanceof Node) {
+                out.appendChild(toElement(object, xml, abbr));
+            } else if (object instanceof Literal) {
+                out.appendChild(xml.createTextNode(((Literal) object).getValue()));
+            } else if (object instanceof NodeReference) {
+                out.appendChild(xml.createTextNode(((NodeReference) object).getIdentifier()));
+            } else {
+                throw new ClassCastException(object.getClass().getName());
+            }
+        }
+
+        for (Entry<String, Collection<ObjectType>> statement : bagElements.entrySet()) {
+            String elementURL = statement.getKey();
+            Element predicate = xml.createElement(abbr.abbreviateElement(RDF.PREDICATE.getIdentifier()));
+            predicate.setAttribute(abbr.abbreviateAttribute(RDF.PREDICATE.getIdentifier(), RDF.VALUE.getIdentifier()),
+                elementURL);
+            predicate.setAttribute(
+                abbr.abbreviateAttribute(RDF.PREDICATE.getIdentifier(), RDF.PROPERTY.getIdentifier()),
+                RDF.BAG.getIdentifier());
+            for (ObjectType object : statement.getValue()) {
+                predicate.appendChild(toElement(object, xml, abbr));
+            }
+            out.appendChild(predicate);
+        }
+    }
+
+    /**
+     * Transforms a node to a DOM document.
+     *
+     * @param node
+     *            node to transform
+     * @param namespaces
+     *            An external Mapping of namespaces to abbreviations. May be
+     *            null.
+     * @return the DOM document
+     */
+    private static Document toDocument(Node node, Map<String, String> namespaces) {
+        try {
+            Namespaces abbr = namespaces instanceof Namespaces ? (Namespaces) namespaces : new Namespaces(namespaces);
+            String qualifiedName = abbr.abbreviateElement(getTypeOrDefault(node));
+            Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().getDOMImplementation()
+                    .createDocument(Namespaces.namespaceOfForXMLFile(getTypeOrDefault(node)), qualifiedName, null);
+            Element root = document.getDocumentElement();
+            if ((node instanceof Node) && (node instanceof IdentifiableNode)) {
```

Comment:
Condition check node instanceof Node should be removed, and if any check is needed here, it is null check (Objects.nonNull(node)).